### ISSUE TITLE
Using SHAP values to estimate value add of a feature

### ISSUE THREAD

**Issue by ytsaig** [shap/shap#545]
Thank you for all the work on this package, it is a fantastic contribution to the ML community!

I have a conceptual question on whether there's a way to use SHAP values to assess the marginal contribution of a new feature to an existing set of features. To make things concrete: suppose we have a model with 100 features, and I construct a 101th feature. I'm interested in estimating the the value add of the 101th feature to the generalization power of the model. While I don't have a formal definition in mind, here's one (inefficient) way to estimate it: 
1. Train first model on existing features, compute accuracy on validation set (or via CV).
2. Train second model on existing features + 101th feature, compute accuracy on validation set.
3. Use the difference in accuracies as the value add of the 101th feature.

This approach is not scalable, but quite useful in understanding the contribution of features to prediction accuracy. When thinking about using SHAP values to get a similar result, I see two issues:
1. SHAP doesn't know anything about model generalization, by construction. To alleviate that, we can assume that the expected training error is a good estimate of the generalization error, i.e., that a decrease in training loss implies a decrease in validation loss.
2. The bigger issue is that, as pointed out in [#288](https://github.com/slundberg/shap/issues/288#issuecomment-457916528), Shapley values distribute credit among correlated features. For example, if feature 101 is some linear combination of the first 100 features, it will be assigned some non-zero aggregate SHAP value, though by definition it does not contribute to model accuracy.

I'm wondering if you have any suggestions or thoughts on how to calculate feature "value add" using SHAP?

**Comment by detrin** [comment]
@ytsaig Thanks for the issue, it is interesting idea.

The schema you proposed is interesting, but it would answer more the question of how adding one feature will affectcontribution of other features. I don't think that this is goal of shape.

> SHAP doesn't know anything about model generalization, by construction. To alleviate that, we can assume that the expected training error is a good estimate of the generalization error, i.e., that a decrease in training loss implies a decrease in validation loss. 

Yes, it doesn't know, but I believe it is just a tool used to explain black-box methods.

> The bigger issue is that, as pointed out in https://github.com/slundberg/shap/issues/288#issuecomment-457916528, Shapley values distribute credit among correlated features. For example, if feature 101 is some linear combination of the first 100 features, it will be assigned some non-zero aggregate SHAP value, though by definition it does not contribute to model accuracy.

I would argue that using correlated features in models is generally bad practice, unless it is something like image processing and it would take huge amount of time or is not desirable in the end product due to the speed of processing.

> I'm wondering if you have any suggestions or thoughts on how to calculate feature "value add" using SHAP?

The way I see it is to use shap for explanation of already final model. When you are in the stage of making decisions what should be added and what not. The loss function or the target metric should be the only criteria for that.
